libraries/browser-tracker-core/src/tracker/index.ts

```diff
@@ -365,20 +350,40 @@ export function Tracker(
     }
 
     /**
-     * Decorate the querystring of a single link
+     * Create link handler to decorate the querystring of a link (onClick/onMouseDown)
      *
      * @param event - e The event targeting the link
      */
-    function linkDecorationHandler(evt: Event) {
-      const timestamp = new Date().getTime();
-      const elt = evt.currentTarget as HTMLAnchorElement | HTMLAreaElement | null;
-      if (elt?.href) {
-        elt.href = decorateQuerystring(elt.href, '_sp', domainUserId + '.' + timestamp);
-      }
+    function addLinkDecorationHandler(extended: boolean): (evt: Event) => void {
+      const CROSS_DOMAIN_PARAMETER_NAME = '_sp';
+
+      return (evt) => {
+        const elt = evt.currentTarget as HTMLAnchorElement | HTMLAreaElement | null;
+        let crossDomainParameterValue;
+        const timestamp = new Date().getTime();
+        if (extended) {
+          const extendedLinkData: ExtendedCrossDomainLinker = {
+            domainUserId,
+            userId: businessUserId || undefined,
+            sessionId: memorizedSessionId,
+            sourceId: configTrackerSiteId,
+            sourcePlatform: configPlatform,
+            timestamp,
+            reason: collectCrossDomainLinkText ? elt?.textContent?.trim() : undefined,
+          };
+          crossDomainParameterValue = encodeURIComponent(btoa(JSON.stringify(extendedLinkData)));
```

Comment:
Question: I think we already discussed this but tbh I'm not 100% clear on what is the reason for using base64 encoding on the value. The entropy of the parameter will be the same, just that we won't see a UUID in there but a different set of characters – is there reason to believe that browsers will be less likely to remove that? I am just a bit worried that we are adding unnecessary complexity here.

Reply:
Excellent question, about the B64, the reason I added it was that I imagine internal apps, or in the future some more privacy-friendly browsers as well, might implement a 'strip email' filter from URL params.
Since `userId` will be a part of the new payload, and in some cases the user ID might be set as the email, I feel that this will give us some extra room.

WDYT ?